### ISSUE TITLE
(V2.0) Audit failure (Missing loaded method) when class moved between symbol dictionaries (with or without class modfifications)

### ISSUE THREAD

**Issue by dalehenrich** [GemTalk/Rowan#587]
After fixing #586, the [audit failures are now showing up in the #586 tests](https://github.com/GemTalk/Rowan/issues/586#issuecomment-598960894):
```
==============Auditing  project RowanSample9
==============Auditing  package RowanSample9-Core
---------------RowanSample9Class1>>Missing loaded method>>foo
```
Back in January, I had noticed that a number of other tests (listed below) had started failing with "missing loaded method" audit failures:
```
RwRowanSample4Test debug: #testIssue185_move_class_to_symbolDict
RwRowanSample4Test debug: #testIssue185_move_class_to_symbolDict_A_v20
RwRowanSample4Test debug: #testIssue185_move_newClassVariable_to_symbolDict
RwRowanSample4Test debug: #testIssue185_move_newClassVersion_to_symbolDict
RwRowanSample4Test debug: #testIssue185_rename_package_move_class
RwRowanSample4Test debug: #testIssue185_rename_package_move_classVariable
RwRowanSample4Test debug: #testIssue185_rename_package_move_newClassVersion
RwRowanSample4Test debug: #testIssue185_simple_package_rename
```

Note that these are all RowanSample4 issues, so now that I'm porting RowanSample4 issues to V2, it's a good time to address them, especially since the bug is clearly showing up in V2.

**Comment by dalehenrich** [comment]
The loaded package was moved to the correct registry, the loaded class was moved to the correct registry, the loaded method was removed from the old registry and NOT added to the new registry.

**Comment by dalehenrich** [comment]
Turns out that I **do** have logic to handle moved classes and this is definitely a case of a moved class, but it is currently slipping between the cracks, because a moved class is currently defined as a class that is moving between packages, but the package is moving between symbol dictionaries and dragging everything in it along ...

**Comment by dalehenrich** [comment]
I'm going to take a crack at changing the logic in `RwGsSymbolDictionaryRegistry_Implementation>>movePackage:to:classesWithNewVersions: instance:` but we will have to exclude `movedClasses`, since this is the boy that is not doing it's job.

**Comment by dalehenrich** [comment]
Will need a test case that combines a symbol dictionary change with a move of a class to/from the symbol dictionary -- where `movedClasses` includes the key of a class that is being moved ... not sure that it is possible to create this combination

**Comment by dalehenrich** [comment]
Hmm, the loaded method is not in the loaded method registry nor is it in the loaded class...

**Comment by dalehenrich** [comment]
...there is a movedPackage, but the class is not in the movedClassesMap, and that is why it is not handled correctly ... if the class is not in the movedClassesMap, but the package is a movedPackage, I think that the classes in the package need to be added to the movedClassesMap ...

**Comment by dalehenrich** [comment]
At this point the proposed fix is patching up the loaded methods correctly, however, the loadedPackages have not been moved to the new symbol dictionary as they should have. Thanks to #588, we are now getting the following audit error:
```
==============Auditing  project RowanSample9
==============Auditing  package RowanSample9-Tests
---------------RowanSample9-Tests>>The loaded class symbol dictionary name 'RowanSample9_2' does not match the loaded package symbol dictionary name 'RowanSample9_1'
==============Auditing  package RowanSample9-Core
---------------RowanSample9-Core>>The loaded class symbol dictionary name 'RowanSample9_2' does not match the loaded package symbol dictionary name 'RowanSample9_1'
```

**Comment by dalehenrich** [comment]
The loaded packages are now being moved to the correct registry, but the symbol dictionary property for the package is not properly updated:
```
==============Auditing  project RowanSample9
==============Auditing  package RowanSample9-Tests
---------------RowanSample9-Tests>>The loaded package 'RowanSample9-Tests' is registered in the wrong symbol dictionary ('RowanSample9_2'). It is expected to be registered in 'RowanSample9_1'
---------------RowanSample9-Tests>>The loaded class symbol dictionary name 'RowanSample9_2' does not match the loaded package symbol dictionary name 'RowanSample9_1'
==============Auditing  package RowanSample9-Core
---------------RowanSample9-Core>>The loaded package 'RowanSample9-Core' is registered in the wrong symbol dictionary ('RowanSample9_2'). It is expected to be registered in 'RowanSample9_1'
---------------RowanSample9-Core>>The loaded class symbol dictionary name 'RowanSample9_2' does not match the loaded package symbol dictionary name 'RowanSample9_1'
```

**Comment by dalehenrich** [comment]
Additional test outstanding regressions that have shown up, since b6fd037 and need to be passing before work on this issue is complete:
- [ ] RwRowanSample9Test>>testIssue493_RowanSample2
- [ ] RwRowanSample9Test>>testIssue586_move_class_to_symbolDict_1
- [ ] RwRowanSample9Test>>testIssue586_move_class_to_symbolDict_3
- [ ] RwMoveTest>>testProjectSet_move_class_between_existing_packages_and_symbolDict
- [ ] RwBrowserToolApiTest>>testNewClassVersion_260_unchanged_extension_method_protocol
- [ ] RwBrowserToolApiTest>>testNewClassVersion_multi_project_unchanged_extension_method_protocol
- [ ] RwHybridBrowserToolTest>>testHybridNewClassVersionWithSubclassesAndMethodsC
- [ ] RwHybridBrowserToolTest>>testHybridProjectLoad
- [ ] RwRowanProjectIssuesTest>>testIssue114_classComment
- [ ] RwRowanProjectIssuesTest>>testIssue114_classCommentChange_1
- [ ] RwRowanProjectIssuesTest>>testIssue313_4
- [ ] RwRowanProjectIssuesTest>>testIssue313_5
- [ ] RwRowanProjectIssuesTest>>testIssue313_6
- [ ] RwRowanProjectIssuesTest>>testIssue467_new_version_class_with_subclasses_5
- [ ] RwRowanProjectIssuesTest>>testIssue467_new_version_class_with_subclasses_6
- [ ] RwRowanProjectIssuesTest>>testIssue91
- [ ] RwRowanSample9Test>>testSpec_0002_to_0003
- [ ] RwRowanSample9Test>>testSwitchRepository

**Comment by dalehenrich** [comment]
I've gotten down to the last few regressions:
- RwRowanSample9Test>>testIssue493_RowanSample2
- RwRowanSample9Test>>testSpec_0002_to_0003

and `RwRowanSample9Test>>testIssue493_RowanSample2` is likely to break other tests, since it's failure is a side-effect of the fix for Issue #313... so I will reopen #313, attach the notes about the failing tests and work on porting the remaining RowanSample4 tests before focusing on more difficult bug fixes.
